### DISCUSSION TITLE
skywalking service diff

### DISCUSSION THREAD

**Discussion by wang1000001000** [apache/skywalking#11467] [Q&A]
How does skywalking distinguish whether a service is a client or a server？
What is the difference between indexes service_relation_server_side and service_relation_client_side?
It's a bit confusing here.

**Comment by wang1000001000** [comment]
I think determining whether a service is a server or a client is relative. If a service calls two services and one is a service provider, then it is in the service_ Relationship_ Server_ On the side, if it is a service consumer, it is reflected in the service_ Relationship_ Client_ Side, I don't know if that's right or not.

**Comment by wu-sheng** [comment]
Yes and No. I am not sure your point.
Generally, this is about which side is a relationship detected.

If you have agent at both sides, you have detected relationship twice.

**Reply by wang1000001000** [reply]
I understand what you mean by whether a call exists in a service_relation_client_side index or  service_relation_server_side index depends on which end STAM will detect, rather than the call being detected twice. It not exists in two indexes at the same time. I don't know if I understand it from this perspective, right

**Reply by wu-sheng** [reply]
It does exist at both sides if you have agents at both sides.

**Reply by wang1000001000** [reply]
I understand. Thank you for your answer.
